### PR TITLE
Update api.md

### PR DESCRIPTION
**Pull request by knottAutodesk**
The link to `The Life of an Entity` was broken.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by knottAutodesk** [comment]
I could not find an appropriate package for this change using `yarn changeset`.

File: docs/features/software-catalog/api.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ with a `Bearer` token, which should then be the Backstage token returned by the
 These are the endpoints that deal with reading of entities directly. What it
 exposes are final entities - i.e. the output of all processing and the stitching
 process, not the raw originally ingested entity data. See [The Life of an
-Entity](life-of-an-entity.md) for more details about this process and
+Entity](https://backstage.io/docs/features/software-catalog/life-of-an-entity) for more details about this process and
```
Comment (awanlin) [inline docs/features/software-catalog/api.md:35]: You hit an odd one, I've seen this before and I think the correct fix is getting this all on one line

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @knottAutodesk, I've suggested a different way to fix this. Also, as this is only documentation and it's not part of a package you don't need a changeset 👍

File: docs/features/software-catalog/api.md (hunk 1 above)
Comment (awanlin) [reply docs/features/software-catalog/api.md:35]: For example like this:

image

Comment (knottAutodesk) [reply docs/features/software-catalog/api.md:35]: Ah yes, I will update this PR.

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍

**Comment by awanlin** [comment]
Thanks again @knottAutodesk 🚀

**Pull request merged**